unistd/crypt.c

```diff
@@ -176,6 +178,15 @@ static void sha1_process(sha1_ctx *ctx, const char *msg)
 
 	if (ctx->ext)
 		sha1_processBlock(ctx);
+
+	/* Fix byte order */
+	for (size_t i = 0; i < sizeof(ctx->hash.h) / sizeof(ctx->hash.h[0]); i++) {
+		uint32_t tmp = ctx->hash.h[i];
+		ctx->hash.res[i * 4] = (tmp & 0xffu);
+		ctx->hash.res[i * 4 + 1] = (tmp & 0xff00u) >> 8;
+		ctx->hash.res[i * 4 + 2] = (tmp & 0xff0000u) >> 16;
+		ctx->hash.res[i * 4 + 3] = (tmp & 0xff000000u) >> 24;
```

Comment:
Maybe:
```c
uint32_t __builtin_bswap32 (uint32_t x)
```

Reply:
I've already merged it - but regarding your comment, I'd prefer to write endian-independent code where possible.

Reply:
This gets optimized on LE systems anyway.